packages/core/src/compile.ts

```diff
@@ -124,30 +120,24 @@ function serialiseObjectToType(v: any) {
   let result = '';
 
   for (const [key, value] of Object.entries(v)) {
-    if (value && typeof value === 'object') {
-      result += `'${encodeWithinSingleQuotes(key)}': ${serialiseObjectToType(
-        value,
-      )},`;
-    } else {
-      result += `'${encodeWithinSingleQuotes(key)}': ${value},`;
-    }
+    result += `${JSON.stringify(key)}: ${
+      value && typeof value === 'object' ? serialiseObjectToType(value) : value
+    },`;
   }
 
   return `{ ${result} }`;
 }
 
-const banner = `// This file is automatically generated by Vocab.\n// To make changes update translation.json files directly.`;
-
-const serializeModuleImports = (imports: Set<string>, moduleName: string) => {
-  if (imports.size === 0) {
+const serializeTypeImports = (
+  imports: Set<string> | Array<string>,
+  moduleName: string,
+) => {
+  const importNames = Array.from(imports);
```

Comment:
I don't see a change where `serializeTypeImports` is being passed an array, so IMO the type should stay as-is. Alternatively you could do `const importNames = new Set(imports);` to handle the array.

Reply:
I had a version where `serializeTypeImports` was used more than once, but not any more.